libcudacxx/include/cuda/std/string_view

```diff
@@ -0,0 +1,800 @@
+  }
+
+  template <class _Range>
+    requires(!is_same_v<remove_cvref_t<_Range>, basic_string_view> && _CUDA_VRANGES::contiguous_range<_Range>
+             && _CUDA_VRANGES::sized_range<_Range> && is_same_v<_CUDA_VRANGES::range_value_t<_Range>, _CharT>
+             && !is_convertible_v<_Range, const _CharT*>
+             && (!requires(remove_cvref_t<_Range>& __d) { __d.operator basic_string_view<_CharT, _Traits>(); }))
+  constexpr explicit _LIBCUDACXX_HIDE_FROM_ABI basic_string_view(_Range&& __r)
+      : __data_(_CUDA_VRANGES::data(__r))
+      , __size_(_CUDA_VRANGES::size(__r))
+  {}
+#endif
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const_iterator begin() const noexcept
+  {
+    return cbegin();
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const_iterator end() const noexcept
+  {
+    return cend();
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const_iterator cbegin() const noexcept
+  {
+    return const_iterator(__data_);
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const_iterator cend() const noexcept
+  {
+    return const_iterator(__data_ + __size_);
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const_reverse_iterator rbegin() const noexcept
+  {
+    return const_reverse_iterator(cend());
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const_reverse_iterator rend() const noexcept
+  {
+    return const_reverse_iterator(cbegin());
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const_reverse_iterator crbegin() const noexcept
+  {
+    return const_reverse_iterator(cend());
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const_reverse_iterator crend() const noexcept
+  {
+    return const_reverse_iterator(cbegin());
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr size_type size() const noexcept
+  {
+    return __size_;
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr size_type length() const noexcept
+  {
+    return __size_;
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr size_type max_size() const noexcept
+  {
+    return numeric_limits<size_type>::max() / sizeof(value_type);
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr bool empty() const noexcept
+  {
+    return __size_ == 0;
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const_reference operator[](size_type __pos) const
+  {
+    _CCCL_ASSERT(__pos < size(), "string_view[] index out of bounds");
+    return __data_[__pos];
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const_reference at(size_type __pos) const
+  {
+    if (__pos >= size())
+    {
+      _CUDA_VSTD::__throw_out_of_range("string_view::at");
+    }
+    return __data_[__pos];
+  }
+
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const_reference front() const
+  {
+    _CCCL_ASSERT(!empty(), "string_view::front(): string is empty");
```

Comment:
For the sake of debug performance I would love to inline calls to single statement functions like `empty`